### PR TITLE
chore(sematic-release): Set up semantic-release with Hackage plugin 

### PR DESCRIPTION
**Pull request by dalejo96**
I have created the semantic-release support for this repo with the following:

- Add package.json to manage semantic-release and semantic-release-hackage versions.
- Use latests yarn
- Add .nvmrc file
- Update the CI workflow with the node installation
- Add releaserc file for plugin configuration

We need to change to a conventional commit format PR #41 for testing this workflow.

Please, review

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,12 +14,48 @@ concurrency:
   cancel-in-progress: true
```
Comment (Alex0jk) [inline .github/workflows/release.yml:14]: I think we can all delete the draft release action.
We wont need it with semantic release.

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,15 @@ cabal.project.local
...
+.vscode
+
+# NodeJS
+node_modules/
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (Alex0jk) [inline .gitignore:35]: The usual, new line at the end of file.

**Review by Alex0jk — COMMENTED** [review]
Looks good,
just left a suggestion to do some more deletions

**Comment by Alex0jk** [comment]
@dalejo96 is the `yarn-4.1.0.cjs` file needed here?

**Review by JoseLion — COMMENTED** [review]
Looks good so far. I left a few comments, let me know if you have any questions 🙂

**Comment by dalejo96** [comment]
> @dalejo96 is the `yarn-4.1.0.cjs` file needed here?

Yes this is required by yarn

**Review by Alex0jk — APPROVED** [review]
Looks good

**Review by Alex0jk — APPROVED** [review]

**Pull request merged**